posthog/api/test/test_organization_feature_flag.py

```diff
@@ -44,3 +47,87 @@ def test_get_feature_flag_unauthorized(self):
         response = self.client.get(url)
 
         self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class TestOrganizationFeatureFlagCopy(APIBaseTest, QueryMatchingTest):
+    def setUp(self):
+        self.team_1 = self.team
+        self.team_2 = Team.objects.create(organization=self.organization)
+
+        self.feature_flag_key = "copied-flag-key"
+        self.feature_flag_to_copy = FeatureFlag.objects.create(
+            team=self.team_1,
+            created_by=self.user,
+            key=self.feature_flag_key,
+            filters={"groups": [{"rollout_percentage": 65}]},
+        )
+
+        super().setUp()
+
+    @snapshot_postgres_queries
+    def test_copy_feature_flag_success(self):
+        url = f"/api/organizations/{self.organization.id}/feature_flags/copy_flags"
+        data = {
+            "feature_flag_key": self.feature_flag_to_copy.key,
+            "from_project": self.feature_flag_to_copy.team_id,
+            "target_project_ids": [self.team_2.id],
+        }
+        response = self.client.post(url, data)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn("success", response.json())
+        self.assertIn("failed", response.json())
+
+        # copied flag is present in the response
+        self.assertEqual(self.feature_flag_key, response.json()["success"][0]["key"])
+
+        # copied flag is present in the DB
+        copied_flag = FeatureFlag.objects.filter(team=self.team_2, key=self.feature_flag_key).exists()
+        self.assertTrue(copied_flag)
+
+    def test_copy_feature_flag_missing_fields(self):
+        url = f"/api/organizations/{self.organization.id}/feature_flags/copy_flags"
+        data = {}
+        response = self.client.post(url, data)
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertIn("error", response.json())
+
+    def test_copy_feature_flag_nonexistent_key(self):
+        url = f"/api/organizations/{self.organization.id}/feature_flags/copy_flags"
+        data = {
+            "feature_flag_key": "nonexistent-key",
+            "from_project": self.team_1.id,
+            "target_project_ids": [self.team_2.id],
+        }
+        response = self.client.post(url, data)
+
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
```

Comment:
hmm, I think this should return 400, resource was found, but with invalid user input?